gui/pages/disk_config.go

```diff
@@ -88,7 +88,8 @@ func (disk *DiskConfig) advancedButtonToggled() {
 	disk.encryptCheck.SetSensitive(storage.AdvancedPartitionsRequireEncryption(disk.model.TargetMedias))
 	disk.encryptCheck.SetActive(false) // Force off for Advance as not support yet
 
-	results := storage.DesktopValidateAdvancedPartitions(disk.model.TargetMedias)
+	results := storage.DesktopValidateAdvancedPartitions(disk.model.TargetMedias,
+		disk.model.LegacyBios, disk.model.SkipValidationSize)
```

Comment:
Looks like DiskConfig type has access to a model struct from which we can extract these bits of info. Maybe just pass the DiskConfig  and we can extract all these values in DesktopValidateAdvancedPartitions itself

Reply:
The storage.DesktopValidateAdvancedPartitions is used by CLI, TUI, and the GUI, so we can't pass a structure unique to just one of those.
We can't just pass a pointer to the model or we get recursive inclusions.

Reply:
Cant we only pass pointer to the model(only this parameter) and unpack things from it if we need it later?

Reply:
> Cant we only pass pointer to the model(only this parameter) and unpack things from it if we need it later?

We can't just pass a pointer to the model or we get recursive inclusions; as stated above.